From c88c39ea4c52956fd9301d62e47f435653939f9e Mon Sep 17 00:00:00 2001
From: Mathieu Bivert <mathieu.bivert@intel.com>
Date: Tue, 30 Apr 2013 09:48:23 +0200
Subject: [PORT FROM MAIN][BT] crashtool: log fw loading failure

BZ: 104695

Change-Id: I3e58003c5eb1387734f7ed2ba8c20f5f2b833a9c
Orig-Change-Id: If44e9f07232da18c56bf0940b08467991937b893
Feature: BT
Category: Improvement
Purpose: AOSP+
Origin: Internal
Upstream-candidate: no
Signed-off-by: Mathieu Bivert <mathieu.bivert@intel.com>
Reviewed-on: http://android.intel.com:8080/105486
Reviewed-by: cactus <cactus@intel.com>
Reviewed-by: Gros, Julien <julien.gros@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: buildbot <buildbot@intel.com>
Tested-by: buildbot <buildbot@intel.com>
---
 libbt/src/hardware.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/libbt/src/hardware.c b/libbt/src/hardware.c
index ba970f1..c9934df 100644
--- a/libbt/src/hardware.c
+++ b/libbt/src/hardware.c
@@ -45,6 +45,8 @@
 #include "userial_vendor.h"
 #include "upio.h"
 
+#include <lct.h>
+
 /******************************************************************************
 **  Constants & Macros
 ******************************************************************************/
@@ -870,6 +872,7 @@ check_local_name:
                         if ((hw_cfg_cb.fw_fd = open(tmp_path, O_RDONLY)) == -1)
                         {
                             ALOGE("vendor lib preload failed to open [%s]", tmp_path);
+                            lct_log(CT_EV_STAT, "cws.bt", "fw_error", 0, tmp_path);
                         }
                     }
                     else
@@ -877,6 +880,7 @@ check_local_name:
                         ALOGE( \
                         "vendor lib preload failed to locate firmware patch file" \
                         );
+                        lct_log(CT_EV_STAT, "cws.bt", "fw_error", 0, tmp_path);
                     }
                 }
 
@@ -1070,6 +1074,7 @@ check_local_name:
     if (is_proceeding == FALSE)
     {
         ALOGE("vendor lib fwcfg aborted!!!");
+        lct_log(CT_EV_STAT, "cws.bt", "fw_cfg", 0);
         if (bt_vendor_cbacks)
         {
             if (p_buf != NULL)
-- 
1.7.4.1

